src/test/custom_loader_test.cpp

```diff
@@ -1018,4 +1084,249 @@ TEST_F(TestCustomLoader, CustomLoaderMultipleLoaderWithSameLoaderName) {
     performPredict("dummy", 1, request);
 }
 
+TEST_F(TestCustomLoader, CustomLoaderBlackListingModel) {
+    // Copy dummy model to temporary destination
+    std::filesystem::copy("/ovms/src/test/dummy", cl_model_1_path, std::filesystem::copy_options::recursive);
+    // Replace model path in the config string
+    std::string configStr = custom_loader_config_model_blacklist;
+    configStr.replace(configStr.find("/tmp/test_cl_models"), std::string("/tmp/test_cl_models").size(), cl_models_path);
+
+    // Create config file
+    std::string fileToReload = cl_models_path + "/cl_config.json";
+    createConfigFileWithContent(configStr, fileToReload);
+
+    ovms::ModelManager& manager = ovms::ModelManager::getInstance();
+    manager.startFromFile(fileToReload);
+
+    ModelServiceImpl s;
+    tensorflow::serving::GetModelStatusRequest req;
+    tensorflow::serving::GetModelStatusResponse res;
+
+    auto model_spec = req.mutable_model_spec();
+    model_spec->Clear();
+    model_spec->set_name("dummy");
+    model_spec->mutable_version()->set_value(1);
+
+    ::grpc::Status ret = test_PerformModelStatusRequestForCustomLoader(s, req, res);
+
+    const tensorflow::serving::GetModelStatusResponse response_const = res;
+    std::string json_output;
+    Status error_status = GetModelStatusImpl::serializeResponse2Json(&response_const, &json_output);
+    ASSERT_EQ(error_status, StatusCode::OK);
+    EXPECT_EQ(json_output, expected_json_available);
+
+    // copy status file
+    std::string status_file_path = cl_model_1_path + "1";
+    std::string status_str = "DISABLED";
+    std::string status_file = status_file_path + "/dummy.status";
+    createConfigFileWithContent(status_str, status_file);
+
+    sleep(10);

```

Comment:
Do we really need this sleep? This significantly increases our build time. Does this test config reload behavior? If so, in test_utils.hpp there is a class `ConstructorEnabledModelManager` which has `loadConfig` method exposed for loading configuration file without watch thread. This way you do not need to wait for OVMS to have separate thread load it for you.